packages/components/src/internal/components/productnavigation/ProductProjectsDrawer.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { FC, memo, useEffect, useState } from 'react';
+import { getServerContext } from '@labkey/api';
+
+import { Container, Alert, buildURL } from '../../..';
+
+import { ProductModel } from './models';
+import { LK_DOC_DEFAULT, PRODUCT_DOC_MAP } from './constants';
+import { ProductClickableItem } from './ProductClickableItem';
+
+interface ProductAppsDrawerProps {
+    product: ProductModel;
+    projects: Container[];
+    onClick: (productId: string, project?: Container) => void;
+}
+
+export const ProductProjectsDrawer: FC<ProductAppsDrawerProps> = memo(props => {
+    const { product, projects, onClick } = props;
+    const { user } = getServerContext();
+
+    const [transition, setTransition] = useState<boolean>(false);
+    useEffect(() => {
+        // use setTimeout so that the "left" property will change and trigger the transition
+        setTimeout(() => setTransition(true), 10);
+    }, []);
```

Comment:
Yeah, odd indeed. Any maybe I'm just using the css transition incorrectly, but it seems that the transition (slide left in this case) will only happen when a property changes in the CSS. So in this case I needed to first render the component so it gets a css prop of `left: 400px` and then add a class using this transition state to the component so that the css prop changes to `left: 0`. And that prop change triggers my ` transition: left .1s ease`. I'll try coming up with a better variable name.